### PR TITLE
RSpec and other test tools init

### PR DESCRIPTION
**Pull request by DandyCV**

### CODE REVIEW

File: .overcommit.yml [hunk 1]
```diff
@@ -69,3 +69,13 @@ PreCommit:
       - '**/*.ru'
       - '**/Gemfile'
       - '**/Rakefile'
+
+CommitMsg:
+  CapitalizedSubject:
+    enabled: false
+
+PrePush:
```
Comment (bestwebua) [inline .overcommit.yml]: Needless.

File: .rspec [hunk 2]
```diff
@@ -0,0 +1 @@
+--require spec_helper
```
Comment (bestwebua) [inline .rspec:1]: Add additional formatter please:
```--format documentation```

File: .simplecov [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+SimpleCov::Formatter::LcovFormatter.config.report_with_single_file = true
+SimpleCov.formatters = SimpleCov::Formatter::MultiFormatter.new(
+  [
+    SimpleCov::Formatter::HTMLFormatter,
+    SimpleCov::Formatter::LcovFormatter
+  ]
+)
+
+SimpleCov.minimum_coverage(90)
```
Comment (bestwebua) [inline .simplecov]: Increase it to 100%.

File: Gemfile [hunk 4]
```diff
@@ -20,6 +20,16 @@ gem 'puma', '~> 5.0'
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: %i[mri mingw x64_mingw]
```
Comment (bestwebua) [inline Gemfile]: Can remove it.

File: spec/models/application_record_spec.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ApplicationRecord, type: :model do
+  it { expect(described_class).to be < ActiveRecord::Base }
+  it { expect(described_class.abstract_class).to be(true) }
+end
```
Comment (bestwebua) [inline spec/models/application_record_spec.rb]: Empty line

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+RSpec.configure do |config|
+  config.expect_with :rspec do |expectations|
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+  end
+
+  config.mock_with :rspec do |mocks|
+    mocks.verify_partial_doubles = true
```
Comment (bestwebua) [inline spec/spec_helper.rb:9]: Also add:
```ruby
mocks.allow_message_expectations_on_nil = true
```

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  end
+  config.shared_context_metadata_behavior = :apply_to_host_groups
```
Comment (bestwebua) [inline spec/spec_helper.rb:12]:
```ruby
config.filter_run_when_matching :focus
config.example_status_persistence_file_path = 'spec/examples.txt'
config.disable_monkey_patching!
config.default_formatter = 'doc' if config.files_to_run.one?
config.profile_examples = 10
config.order = :random

Kernel.srand(config.seed)

config.include FactoryBot::Syntax::Methods
config.include ActiveSupport::Testing::TimeHelpers
```

**Review by bestwebua — CHANGES_REQUESTED** [review]

File: .overcommit.yml (hunk 1 above)
Comment (DandyCV) [reply .overcommit.yml]: Removed

File: .rspec (hunk 2 above)
Comment (DandyCV) [reply .rspec:1]: Added

File: .simplecov (hunk 3 above)
Comment (DandyCV) [reply .simplecov]: OK!

File: Gemfile (hunk 4 above)
Comment (DandyCV) [reply Gemfile]: Removed

File: spec/models/application_record_spec.rb (hunk 5 above)
Comment (DandyCV) [reply spec/models/application_record_spec.rb]: Added

File: spec/spec_helper.rb (hunk 6 above)
Comment (DandyCV) [reply spec/spec_helper.rb:9]: Added!

File: spec/spec_helper.rb (hunk 7 above)
Comment (DandyCV) [reply spec/spec_helper.rb:12]: Added!

File: .overcommit.yml [hunk 8]
```diff
@@ -69,3 +69,12 @@ PreCommit:
       - '**/*.ru'
       - '**/Gemfile'
       - '**/Rakefile'
+
+  Undercover:
```
Comment (bestwebua) [inline .overcommit.yml]: All Undercover block needless.

**Review by bestwebua — CHANGES_REQUESTED** [review]

File: .overcommit.yml (hunk 8 above)
Comment (DandyCV) [reply .overcommit.yml]: Removed

**Review by bestwebua — APPROVED** [review]
👍

**Pull request merged**